### PR TITLE
Update readme

### LINKED ISSUE #14: Peer Code Review
Conduct a code review with another student in your group based on the principles from the Code Review Wikipedia article. Focus on:
o Code readability and maintainability.
o Adherence to coding standards.
o Correct use of mocks in testing.
o Identification of potential bugs.
Provide constructive feedback on your partner's code. Ensure feedback is clear, specific, and actionable.
Use tools like GitHub Pull Requests or GitLab Merge Requests for conducting the review (optional).
Submit your written feedback and any changes suggested or made based on the review.

### PR DESCRIPTION
**Pull request by HelenaLykstoft**

### CODE REVIEW

File: ReadMe.md [hunk 1]
```diff
@@ -1,96 +1,141 @@
-# OLA2: API Testing, Coverage, and Benchmarking
- - Group: **TofuBytes**
- - Members: **Isak, Jamie & Helena**
```
Comment (solskinIsak) [inline ReadMe.md:3]: Burde vi ikke beholde vores gruppe medlemmer og gruppenavne eller har vi dem et andet sted ?

File: ReadMe.md [hunk 2]
```diff
@@ -1,96 +1,141 @@
...
+### 9. Reflection on Testing and Code Quality
+- Write a **300-400 word reflection** on:
+    - Impact of static code analysis tools (PMD, JaCoCo).
+    - Importance of mocking in unit testing.
+    - Value of code and software reviews.
+    - Influence of Equivalence Partitioning and Boundary Value Analysis on test design.
+- Discuss how these practices helped in identifying and fixing issues early.
+
+Impact of static code analysis tools:
+- We implemented Qodana, which is a static code analysis tool. This tool helped us identify issues, redundant code or unused code, and helped us fix them before they became a problem. The impact of Qodana was good, because we were able to actually see all the issues we had in our code. For us, this helps us understanding what exactly we need to focus on improving in the code, so we for examplem dont have a lot of redundant or unused code.
```
Comment (solskinIsak) [inline ReadMe.md:101]: Lille stavefejl ved 'example' her

File: ReadMe.md [hunk 3]
```diff
@@ -1,96 +1,141 @@
...
-Throughput: Handling nearly 18 transactions per second shows that the system can process a moderate load efficiently.
-Overall, the test shows excellent system performance with low response times, high throughput, and no errors.
-
-
-## Reflection
-We didn't satisfy all the assignments criteria (or requirements) within the assigned timeslot and in order to still reach the deadline we focused on working with the tools and gathering an understanding for future projects.
-If we were to make this project again, we would make sure we had enough code coverage, since we only made it to 52% of the overall coverage, and make sure to meet all the requirements.
+---
 
+## Tools and Technologies
```
Comment (solskinIsak) [inline ReadMe.md:128]: Vi skal lige have det rettet til her så der kun står de ting vi har brugt

C# 

Moq

Hverken coverlet eller det andet der stod(vu brugte bare riders indbyggede code coverage værktøj)

Osv osv

**Review by solskinIsak — COMMENTED** [review]
Overordnet set er det godt men vi mangler stadig nogle områder og nogle bider der skal rettes lidt til og slettes / skrives 

Der er nok også nogle ting jeg har misset hahah

Men godt arbejde!

File: ReadMe.md [hunk 4]
```diff
@@ -1,96 +1,141 @@
...
+
+
+### 8. Software Review
+- Perform a **Software Review** based on the [Software Review](https://en.wikipedia.org/wiki/Software_review) article:
+    - Functional correctness: Does the code do what it is supposed to do?
+    - Non-functional aspects (performance, maintainability, etc.).
+    - Best practices (code structure, documentation).
+- Submit a **300-word report** reflecting on your findings and improvements.
+
+In relation to functional correctness, the code performs as expected and passes all the test cases. Testing was done using equivalence partitioning and boundary value analysis.
```
Comment (solskinIsak) [inline ReadMe.md]: Vi mangler at få specificeret hvor vi bruger equivalence partitioning & boundy value analysis.

for det er kun i vores validation tests det bliver anvendt / unit tests. det bliver ikke brugt i vores integration_api_tests som reviewers nok ville kommentere på modsiger hvad vi skriver her

File: ReadMe.md [hunk 5]
```diff
@@ -1,96 +1,141 @@
...
+    - Functional correctness: Does the code do what it is supposed to do?
+    - Non-functional aspects (performance, maintainability, etc.).
+    - Best practices (code structure, documentation).
+- Submit a **300-word report** reflecting on your findings and improvements.
+
+In relation to functional correctness, the code performs as expected and passes all the test cases. Testing was done using equivalence partitioning and boundary value analysis. 
+For equivalence partitioning, the inputs were divided into valid and invalid categories, ensuring that the code handled typical cases within these partitions correctly. 
+Additionally, boundary value analysis revealed that the code responds appropriately at the edges of input ranges, such as minimum and maximum allowable values. However, further testing could be beneficial to expand the test coverage, especially for complex input patterns or extreme edge cases.
+We made sure that our software was correct, and that it was easy to maintain. We also made sure that the code was structured in a way that was easy to understand, and that the documentation was clear and concise.
```
Comment (solskinIsak) [inline ReadMe.md]: er ikke sikker på hvad der bliver ment med dokumentation her er det de få kommentarer inde i test casen? 
```suggestion
We made sure that our software was correct, and that it was easy to maintain. We also made sure that the code was structured in a way that was easy to understand.
```

File: ReadMe.md (hunk 4 above)
Comment (solskinIsak) [inline ReadMe.md]:
```suggestion
In relation to functional correctness, the code performs as expected and passes all the test cases. `Validation_test` was added as a way of ensuring our Validation Testing was done using equivalence partitioning and boundary value analysis. 
```

File: ReadMe.md [hunk 6]
```diff
@@ -1,96 +1,141 @@
...
+The code was made to do what it was supposed to do, and if there were anything that were 'too much' or 'too little', we made sure to fix it.
+
+In relation to best practices, we always try to maintain a clean code structure and follow the coding standards. We make sure to name our methods correctly, so everyone knows what the methods actually do.
+We could improve the documentation, and use more comments directly in the code, so that is something we need to work on.
```
Comment (solskinIsak) [inline ReadMe.md]: Godt punkt at nævne vi vil improve dokumentation da vi ikke har noget nuværende, også derfor det var forvirrende længere oppe
```suggestion
We strive to improve documentation, comments is a way we want to reflect that directly in the code.
```

File: ReadMe.md (hunk 2 above)
Comment (solskinIsak) [inline ReadMe.md]:
```suggestion
- We implemented Qodana, which is a static code analysis tool. This tool helped us identify issues, redundant code, unused code amongst other issues. This helped us detect and fix these issues before they became a problem. The impact of Qodana was good, it provided us with good overheard and enabled us to see the issues that required our attention. For us, this helps us gather a greater understanding of what exactly we need to focus on, and to improve upon for future projects.
```

File: ReadMe.md [hunk 7]
```diff
@@ -1,96 +1,141 @@
...
-## Overview
-This project involves building a simple CRUD (Create, Read, Update, Delete) REST API, writing integration tests for the API and try to ensure at least 80% code coverage.
-
-We didnt get to finish this project, because we used a lot of time on the basic setup of the project, and we had some issues with the testing of the API.
-Therefore we only managed to implement swagger, we managed to make integration tests for the API, we managed to look a little bit into the code coverage and we managed to look into JMeter.
+## Deliverables
+- Refined Task Service source code.
+- Unit tests with mocks.
+- PMD (or FxCop) report.
+- JaCoCo (or Coverlet) code coverage report.
```
Comment (solskinIsak) [inline ReadMe.md:120]:
```suggestion
- JaCoCo (or Coverlet) code coverage report.
  - We used the built-in code coverage tool in JetBrains Rider.
```

File: ReadMe.md [hunk 8]
```diff
@@ -1,96 +1,141 @@
...
 
-## Overview
-This project involves building a simple CRUD (Create, Read, Update, Delete) REST API, writing integration tests for the API and try to ensure at least 80% code coverage.
-
-We didnt get to finish this project, because we used a lot of time on the basic setup of the project, and we had some issues with the testing of the API.
-Therefore we only managed to implement swagger, we managed to make integration tests for the API, we managed to look a little bit into the code coverage and we managed to look into JMeter.
+## Deliverables
+- Refined Task Service source code.
+- Unit tests with mocks.
+- PMD (or FxCop) report.
```
Comment (solskinIsak) [inline ReadMe.md:118]:
```suggestion
- PMD (or FxCop) report.
  - We used the built-in functionality of Qodana in Rider
```

File: ReadMe.md [hunk 9]
```diff
@@ -1,96 +1,141 @@
-# OLA2: API Testing, Coverage, and Benchmarking
- - Group: **TofuBytes**
- - Members: **Isak, Jamie & Helena**
+# Task Service with Testing, Code Analysis, and Reviews
+
```
Comment (solskinIsak) [inline ReadMe.md]:
```suggestion
 - Group name: **TofuBytes** Members: **Isak, Jamie & Helena**
```

File: ReadMe.md [hunk 10]
```diff
@@ -1,96 +1,141 @@
...
-Overall, the test shows excellent system performance with low response times, high throughput, and no errors.
-
-
-## Reflection
-We didn't satisfy all the assignments criteria (or requirements) within the assigned timeslot and in order to still reach the deadline we focused on working with the tools and gathering an understanding for future projects.
-If we were to make this project again, we would make sure we had enough code coverage, since we only made it to 52% of the overall coverage, and make sure to meet all the requirements.
+---
 
+## Tools and Technologies
+- **Languages**: Java/C# (or equivalent).
```
Comment (solskinIsak) [inline ReadMe.md]:
```suggestion
- **Languages**: C#.
```

File: ReadMe.md [hunk 11]
```diff
@@ -1,96 +1,141 @@
...
+- **Mocking**: Mockito, EasyMock (Java), or Moq (C#).
```
Comment (solskinIsak) [inline ReadMe.md]:
```suggestion
- **Mocking**: Moq (C#).
```

File: ReadMe.md [hunk 12]
```diff
@@ -1,96 +1,141 @@
...
+- **Static Code Analysis**: PMD (Java) or FxCop (C#).
```
Comment (solskinIsak) [inline ReadMe.md]:
```suggestion
- **Static Code Analysis**: Qodana & Qodana Cloud .
```

File: ReadMe.md [hunk 13]
```diff
@@ -1,96 +1,141 @@
...
+- **Code Coverage**: JaCoCo (Java) or Coverlet (C#).
```
Comment (solskinIsak) [inline ReadMe.md]:
```suggestion
- **Code Coverage**: Built-in Code Coverage tool in JetBrains Rider.
```

**Review by solskinIsak — CHANGES_REQUESTED** [review]
Det her er bare små ændringer i phrasing og steder vi bare manglede lidt tilrettelse men det skal stadig understreges at det er godt arbejde! 

mine ændringer er bare små ting hvor jeg føler jeg havde fomuleret enkelte ting anderledes

**Review by solskinIsak — APPROVED** [review]
